### ISSUE TITLE
Grad backward memory leak?

### ISSUE THREAD

**Issue by hudongloop** [pytorch/pytorch#3824]
Memory keeps growing at every iteration. My pytorch version 0.4.0a0+4d405a4.
```
import torch
from torch.autograd import grad, Variable
from torchvision import models

model = models.resnet50().cuda()
for k in range(20):
    x   = Variable(torch.rand(8, 3, 224, 224).cuda(), requires_grad=True)
    dx, = grad(model(x).sum(), x, create_graph=True)
    y = model(x + dx).sum()
    y.backward()
```

This code is from [Double backward memory leak](https://github.com/pytorch/pytorch/issues/2361). It work well and no memory leak. But use
```
y = dx.mean()
y.backward()
```
the memory leak is happen.

**Comment by zou3519** [comment]
This might be an instance of #3835 -- I think resnet50 includes [multiple conv2d layers](http://pytorch.org/docs/0.2.0/_modules/torchvision/models/resnet.html).

**Comment by zou3519** [comment]
I was able to reproduce your results: using `y = dx.mean()` instead of `y = model(x + dx).sum()` causes a noticeable memory leak. 

Using `y = model(x + dx).sum()` I see a really small increase in memory after each iteration (< 100kb), but this might just be some measurement error.

**Comment by soumith** [comment]
was this already fixed @colesbury ? or is this a duplicate of #3835

**Comment by colesbury** [comment]
It's not fixed yet. I think it's the same issues as 3835. The PR #3970 seems to fix it, but we may want to do something else for v0.3

**Comment by li-roy** [comment]
Closed by #3970
